src/natcap/invest/validation.py

```diff
@@ -596,17 +600,58 @@ def check_spatial_overlap(spatial_filepaths_list,
     return None
 
 
+def timeout(func, *args, timeout=5, **kwargs):
+    """Stop a function after a given amount of time.
+
+    Args:
+        func (function): function to apply the timeout to
+        args: arguments to pass to the function
+        timeout (number): how many seconds to allow the function to run.
+            Defaults to 5.
+
+    Returns:
+        None
+
+    Raises:
+    """
+
+
+    # use a queue to share the return value from the file checking thread
+    # the target function puts the return value from `func` into shared memory
+    message_queue = queue.Queue()
+    def wrapper_func():
+        message_queue.put(func(*args, **kwargs))
+
+    thread = threading.Thread(target=wrapper_func)
+    LOGGER.info(f'Starting file checking thread with timeout={timeout}')
+    thread.start()
+    thread.join(timeout=timeout)
+
+    if thread.is_alive():
+        return ('Validation of this file timed out. If this file is stored in '
+               'a file-streaming service, it may be taking too long to '
+               'download. Try storing it locally.')
+
+    else:
+        LOGGER.info('File checking thread completed.')
+        # get any warning messages returned from the thread
+        return message_queue.get()
+
+
+# checks that open a file could potentially take a long time
```

Comment:
Should this comment be: 
>opening a file could potentially take a long time

Reply:
Thanks for catching that! Rephrased it to make more sense.